File: 07_app_development/MyProject/libtsparser/src/main/java/com/tosmart/xiexuming/libtsparser/descriptor/DescriptorExtension.java
package com.tosmart.xiexuming.libtsparser.descriptor;

/**
 * @author xiexuming
 */
public interface DescriptorExtension {
    /**
     * 解析成serviceDescriptor
     *
     * @param descriptor 普通descriptor
     * @return 返回serviceDescriptor
     */
    ServiceDescriptor getServiceDescriptor(Descriptor descriptor);
}

File: newpaserapp/libtosmartparse/src/main/java/com/example/libtosmartparse/descriptor/DescriptorExtension.kt
package com.example.libtosmartparse.descriptor

/**
 * @author xxm
 * 2022/7/17
 **/
open interface DescriptorExtension {
    /**
     * 解析成serviceDescriptor
     *
     * @param descriptor 普通descriptor
     * @return 返回serviceDescriptor
     */
    fun getServiceDescriptor(descriptor: Descriptor?): ServiceDescriptor?
}